### PR TITLE
Use `put` methods instead of `extend_from_slice` to write to Vec<u8>

### PR DESCRIPTION
**Pull request by danlaine**
Per [this conversation](https://github.com/commonwarexyz/monorepo/pull/271#discussion_r1894373050)

### CODE REVIEW

File: p2p/src/authenticated/actors/tracker/address.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use bytes::BufMut;
+
```
Comment (patrick-ogrady) [inline p2p/src/authenticated/actors/tracker/address.rs]: nit: can you remove this `\n`?

**Review by patrick-ogrady — CHANGES_REQUESTED** [review]
One unnecessary `\n` removal then should be ready 👍

**Review by patrick-ogrady — APPROVED** [review]

**Comment by patrick-ogrady** [comment]
Looks like `fmt` is failing (I think you just need to reorder the imports).

**Comment by danlaine** [comment]
> Looks like `fmt` is failing (I think you just need to reorder the imports).

Argh. Fixed.

**Review by patrick-ogrady — APPROVED** [review]

**Pull request merged**